dev/test/bulk-writer.ts

```diff
@@ -96,62 +87,73 @@ describe('BulkWriter', () => {
     return remove(doc).writes![0];
   }
 
-  function createRequest(requests: api.IWrite[]): BulkWriterRequest {
+  function createRequest(requests: api.IWrite[]): api.IBatchWriteRequest {
     return {
       writes: requests,
     };
   }
 
-  function successResponse(seconds: number): {[key: string]: {}} {
+  function successResponse(seconds: number): api.IBatchWriteResponse {
     return {
-      updateTime: {
-        nanos: 0,
-        seconds,
-      },
-      code: Status.OK,
+      writeResults: [
+        {
+          updateTime: {
+            nanos: 0,
+            seconds,
+          },
+        },
+      ],
+      status: [{code: Status.OK}],
     };
   }
 
-  function failResponse(): {[key: string]: {} | null} {
+  function failResponse(): api.IBatchWriteResponse {
     return {
-      updateTime: null,
-      code: Status.UNAVAILABLE,
+      writeResults: [
+        {
+          updateTime: null,
+        },
+      ],
+      status: [{code: Status.UNAVAILABLE}],
     };
   }
 
-  function createResponse(
-    responses: Array<{[key: string]: {} | null}>
-  ): BulkWriterResponse {
-    const writeResults = responses.map(response => ({
-      updateTime: response.updateTime,
-    }));
-    const status = responses.map(
-      response => ({code: response.code!} as proto.google.rpc.IStatus)
-    );
+  function mergeResponses(
+    responses: api.IBatchWriteResponse[]
+  ): api.IBatchWriteResponse {
     return {
-      writeResults,
-      status,
+      writeResults: responses.map(v => v.writeResults![0]),
+      status: responses.map(v => v.status![0]),
     };
   }
 
   /**
+   *  /**
    * Creates an instance with the mocked objects.
    *
-   * @param checkActiveRequests Whether to check for active requests. If true,
-   * the `activeRequestDeferred` must be manually resolved for the response to
-   * return.
+   * @param options.enforceSingleConcurrentRequest Whether to check that there
+   * is only one active request at a time. If true, the `activeRequestDeferred`
+   * must be manually resolved for the response to return.
+   * @param options.failResponse Whether to fail the request with an error.
    */
   function instantiateInstance(
     mock: RequestResponse[],
-    checkActiveRequests = false
+    options?: {
+      enforceSingleConcurrentRequest?: boolean;
+      failResponse?: boolean;
+    }
   ): Promise<BulkWriter> {
     const overrides: ApiOverride = {
       batchWrite: async request => {
         expect(request).to.deep.eq({
           database: `projects/${PROJECT_ID}/databases/(default)`,
           writes: mock[requestCounter].request.writes,
         });
-        if (checkActiveRequests) {
+        if (options && options.failResponse) {
+          throw new Error('batchWrite failed');
```

Comment:
As per Gil: Please use 'test' somewhere in the error message. This might save us some debugging time later.

Reply:
changed to `mock batchWrite failed in test`